spec/support/matchers/lead_provider_api/can_retrieve_participant_details_from_the_ecf_participants_endpoint.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module Support
+  module CanRetrieveParticipantDetailsFromTheEcfParticipantsEndpoint
+    extend RSpec::Matchers::DSL
+
+    RSpec::Matchers.define :be_able_to_retrieve_the_details_of_the_participant_from_the_ecf_participants_endpoint do |participant_name, participant_type, options = {}|
+      match do |lead_provider_name|
+        user = User.find_by(full_name: participant_name)
+        raise "Could not find User for #{participant_name}" if user.nil?
```

Comment:
this seems to work so leave it. but it looks you've written your own little pattern for assertions thru exceptions as opposed to using patterns defined by rspec. see https://relishapp.com/rspec/rspec-expectations/docs/custom-matchers/define-a-custom-matcher

Reply:
oh i see, you've done above on line 8 then stuck it in a huge block